File: leetcode/java/1528. Shuffle String.java
class Solution {
    public String restoreString(String s, int[] indices) {
        final char[] array = new char[indices.length];
        
        for(int index : indices) {
            array[indices[index]] = s.charAt(index);
        }
        
        return String.valueOf(array);
    }
}

File: leetcode/kotlin/1528. Shuffle String.kt
class Solution {
    fun restoreString(s: String, indices: IntArray): String {
        val array = CharArray(indices.count())
        
        for (index in 0 until indices.count()) {
            array[indices[index]] = s.get(index)
        }
        
        return array.joinToString("")
    }
}
